policy/modules/kernel/files.fc

```diff
@@ -49,6 +49,7 @@ ifdef(`distro_suse',`
 /etc/\.fstab\.hal\..+	--	gen_context(system_u:object_r:etc_runtime_t,s0)
 /etc/blkid(/.*)?		gen_context(system_u:object_r:etc_runtime_t,s0)
 /etc/cmtab		--	gen_context(system_u:object_r:etc_runtime_t,s0)
+/etc/fstab		--	gen_context(system_u:object_r:fstab_t,s0)
```

Comment:
For what it's worth I have been running this PR on top of 5 different machines in enforcing and haven't noticed any breakage after the initial work that went into it.

My initial motivation for introducing this change is to fix the confusion of `etc_t` systemd units. To me this was one of those things that has always given off a, "Hmm, that doesn't seem right" intuition. The added bonus of this is it allows restricting access to the `fstab` file itself, although I agree I don't think that alone has a lot of value.

If you insist I think I'd be OK with dropping this change and instead just editing the comments around `etc_t` systemd units to clarify these are generated by `systemd-fstab-generator` and will have the same type of `/etc/fstab`.

Reply:
After some discussion on IRC about this I want to bring up this info to consider, although I think your point on the lack of much value is still valid here.

In theory a domain with access to edit `etc_t` files and to manage `etc_t` units could create arbitrary systemd units and essentially escalate to `init_t`. Systemd allows adding and starting arbitrary units by enabling a unit file via an absolute path:
```
# vi /etc/test.service
# systemctl enable /etc/test.service
Created symlink /etc/systemd/system/multi-user.target.wants/test.service → /etc/test.service.
# systemctl start test.service
```
You can also take advantage of `systemctl link` to add arbitrary paths to the unit search paths.

However, to my knowledge there aren't any domains in refpolicy currently with both access to edit `etc_t` files and to manage `etc_t` services (other than `sysadm_t` and `unconfined_t`), so as of right now this is effectively a non-issue.

It's also worth mentioning that editing `/etc/fstab` and controlling `etc_t` units also effectively grants access to control arbitrary mounts as long as `init_t` has the required permissions, but this relies on having the same access as mentioned above.

Reply:
In terms of the systemd units, I think it would probably be better to have a type for units in /etc. Then it's cleaner all around, from managing those units and controlling them. That separates it from fstab too.

Reply:
> I think it would probably be better to have a type for units in /etc.

I think that makes sense and covers all of my original concerns. I'll rework this.

Reply:
>I think it would probably be better to have a type for units in /etc.

After thinking on it more, I'm not sure this is what we want. A unit file can exist anywhere on the system, and the corresponding service object just inherits the unit file's type. We could put a unit file in, say, `/mnt` and enable and start it, although the policy doesn't allow this if the unit file's type is wrong.

I think creating a type for generic units in `/etc` to include `fstab` seems like a step in the wrong direction. I think the only supported use case for arbitrary units to exist elsewhere on the system is if they are manually labeled `systemd_unit_t`. `fstab` is special in that it is the only file which is not inherently a unit file but which service objects are created from.

Reply:
@pebenito I left these original changes in this PR for now, but I am more than happy to split them into a separate PR/issue for additional discussion if it means getting these other changes reviewed and merged sooner.